cluster/v1alpha1/types.go

```diff
@@ -321,6 +327,46 @@ type PrioritizerConfig struct {
 	Weight int32 `json:"weight,omitempty"`
 }
 
+// Toleration represents the toleration object that can be attached to a placement.
+// The placement this Toleration is attached to tolerates any taint that matches
+// the triple <key,value,effect> using the matching operator <operator>.
+type Toleration struct {
+	// Key is the taint key that the toleration applies to. Empty means match all taint keys.
+	// If the key is empty, operator must be Exists; this combination means to match all values and all keys.
+	// +optional
+	Key string `json:"key,omitempty"`
```

Comment:
the regex format here should be the same as taint? Why this is optional?

Reply:
> the regex format here should be the same as taint?

Yes.

> Why this is optional?

If key=empty and operator=Exists, it means to match all values and all keys. In another word, the placement tolerates all taints.